### PR TITLE
Create slimmed-down Docker images for use with Conda development

### PR DESCRIPTION
**Pull request by martyngigg**
The new images contain only an operating system and a few tools like curl, latex that are not available/working in Conda.

This also provides a set of ansible scripts to deploy the image and a new container to new machines. README files have been added/provided to describe how to use the new features.

### CODE REVIEW

File: jenkins-node/mantid-builder/readme.md [hunk 1]
```diff
@@ -1,19 +1,87 @@
...
+- [RancherOS](#rancheros). Not recommended as this is no longer supported.
+
+## Cloud VM
+
+### OS deployment
+
+Use the wizard provided by the cloud provider, e.g. Openstack for SCD cloud at ISIS.
+We're running containers so the base operating system does not really matter.
+If in doubt choose the latest non-gui Ubuntu version. When selecting
+the resources for the machine we suggest atleast 8 cores, 32GB RAM and
```
Comment (sf1919) [inline jenkins-node/mantid-builder/readme.md:19]: There needs to be a space between `at` and `least`.

Comment (martyngigg) [reply jenkins-node/mantid-builder/readme.md]: Thanks. I've fixed that and amended my last commit.

File: development/docker/CentOS7Slim.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Arguments allowed to be used in FROM have to come
+# before the first stage
+ARG CPPCHECK_VERSION=2.5
+
+# Import cppcheck. COPY --from cannot used variables.
+# Define a local name
+FROM neszt/cppcheck-docker:${CPPCHECK_VERSION} AS upstream_cppcheck
+
+# Base
```
Comment (martyngigg) [inline development/docker/CentOS7Slim.Dockerfile:9]: It just occurred to me that it's not really obvious why we have picked CentOS.
Could you add a comment that we did this to match with the platform used by [conda-forge](https://github.com/chrisburr/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L142)?

File: development/docker/install_latex.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#Install latex direct from texlive
```
Comment (martyngigg) [inline development/docker/install_latex.sh:1]: It's usual to add a first line to `sh` scripts of the form `#! /bin/bash` to indicate what can be used to run it.

**Review by martyngigg — COMMENTED** [review]
This looks good to me.

Just a couple of minor comments now.

**Review by martyngigg — COMMENTED** [review]
This looks good and is now deployed on the STFC cloud.

**Pull request merged**